web/assets/login-events.js

```diff
@@ -0,0 +1,24 @@
+// get the challenge from out of the HTML
+let sc = document.querySelector("#challenge").dataset.sc
```

Comment:
```suggestion
const sc = document.querySelector("#challenge").dataset.sc
```

Reply:
This and others is not an important change, but `const` is a bit better to specify intent that we don't want these things to be mutated later on. Helps us a tiny bit protecting against cases we don't want to cover.